src/chatClient.cpp

```diff
@@ -1852,6 +1855,7 @@ promise::Promise<void> GroupChatRoom::decryptTitle()
         wptr.throwIfDeleted();
         KR_LOG_ERROR("Error decrypting chat title for chat %s:\n%s\nFalling back to member names.", karere::Id(chatid()).toString().c_str(), err.what());
         makeTitleFromMemberNames();
+        return err;
```

Comment:
This one is arguable, as the fail() executes a fallback (sets the title to a list of member names), but it really depends how the function is intended to be used. Having it fail and set the seems more semantically correct given the name of the function. I.e. decrypt failed -> return error. We have another small issue here - if the function returns a failed promise, it should be handled by the caller, and currently it's not, it's a fire-and-forget operation. If the function returns a failed promise, you will get an "Unhandled promise fail" warning from the promise lib.